packages/vats/src/proposals/probeZcfBundle.js

```diff
@@ -0,0 +1,74 @@
+import { E } from '@endo/far';
+import { makeStorageNodeChild } from '@agoric/internal/src/lib-chainStorage.js';
+
+const oldZCFBundleId =
+  'b1-039c67a6e86acfc64c3d8bce9a8a824d5674eda18680f8ecf071c59724798945b086c4bac2a6065bed7f79ca6ccc23e8f4a464dfe18f2e8eaa9719327107f15b';
+
+// verify that Zoe remembers the zcfBundleCap across upgrades
+const probeZcfBundleCap = async (
+  {
+    consume: {
+      vatAdminSvc,
+      walletFactoryStartResult,
+      provisionPoolStartResult,
+      chainStorage,
+      walletBridgeManager: walletBridgeManagerP,
+      vatStore,
+    },
+  },
+  options,
+) => {
+  console.log('probeZcfBundle start');
+  const { zoeRef, zcfRef, walletRef } = options.options;
```

Comment:
I'm still confused how these options can get passed in. The way things are structured now, the submission is prepared before a3p proposals are run. However I think this proposal does not actually need to generate any bundle, so the submission could be manually written, with the template mechanism used to fill the probed bundle refs?

Reply:
I think you're right. My current belief is that I want to cite the Zoe and ZCF bundles that will have been included in upgrade-15. I want to do a null upgrade of Zoe to trigger reversion to a stored ZCF bundle. If I can then do a null upgrade to a contract vat (e.g. WalletFactoryI then I can compare the ZCF bundle used to the ZCF bundleID just used in upgrade-15.
So I'm looking for how I can find out those bundleIDs to use with the new (very clean!) template mechanism.

Reply:
@turadg [added a check](https://github.com/Agoric/agoric-3-proposals/pull/59/files) so if the bundle has already been installed, it doesn't do it again. Including these bundle IDs in the manifest is necessary so  I can have them to request an upgrade.

Reply:
I don't understand how that's relevant. I actually don't understand why this test needs to install any new bundle ?

Reply:
I mean I understand how one option is to do a full generation of core eval scripts and bundles, but I thought from the description we just wanted to "restart" the contracts. I suppose we may want to upgrade ZCF since the test checks it uses the new version

Reply:
restarting the contracts would be fine.

When I used `restartContract()` without a bundleID, in place of `E(walletAdminFacet).upgradeContract(walletRef.bundleID, privateArgs);`, I got an error from the kernel that implied that it was trying to upgrade to an earlier version of the contract. `checkAndUpdateFacetiousness()` complained that I was trying to remove a facet, which would only make sense if walletFactory was trying to upgrade from a relatively modern version to the version on chain. I'll file a bug to investigate `restartContract()`.

Since the only productive way to upgrade the wallet factory was using `upgradeContract()`, I needed a bundleId.

With ZCF, I need to pass the bundleID to `E(zoeConfigFacet).updateZcfBundleId(zcfRef.bundleID);` to tell Zoe to use a particular version of ZCF when upgrading the contracts. The point of this test is to validate that #8807 correctly fixes Zoe to store the `ZCFBundleID`. Before that PR, it clearly was not.

Reply:
With the fix in commit `5a30b2d877`, I was able to revert to `restartContract` for the second restart, but the first still needs to be an upgrade (in the test) so that we can tell whether it is using the original bundleId or the newly provided one.